src/test/java/jewellery/inventory/unit/service/ProductServiceTest.java

```diff
@@ -0,0 +1,264 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.util.*;
+
+@ExtendWith(MockitoExtension.class)
+class ProductServiceTest {
+
+  @InjectMocks private ProductService productService;
+  @Mock private ProductRepository productRepository;
+  @Mock UserMapper userMapper;
+  @Mock ProductMapper productMapper;
+  @Mock UserRepository userRepository;
+  @Mock ResourceRepository resourceRepository;
+  @Mock ResourceInUserRepository resourceInUserRepository;
+  @Mock ResourceInProductRepository resourceInProductRepository;
+  @Mock ResourceInUserService resourceInUserService;
+
+  private User user;
+  private Product testProduct;
+  private Resource pearl;
+  private ResourceInUser resourceInUser;
+  private ProductRequestDto productRequestDto;
+  private ResourceQuantityRequestDto resourceQuantityRequestDto;
+
+  @BeforeEach
+  void setUp() {
+
+    user = UserTestHelper.createTestUserWithRandomId();
+    pearl = ResourceTestHelper.getPearl();
+    resourceInUser = createResourceInUser(user, pearl);
+    testProduct = createTestProduct(user, pearl);
+    resourceQuantityRequestDto = createResourceQuantityRequestDto(pearl);
+    productRequestDto = createProductRequestDto(user, resourceQuantityRequestDto);
+  }
+
+  @Test
+  void testDeleteProductContainsProductBreakDown() {
+    testProduct.setProductsContent(List.of(new Product(), new Product(), new Product()));
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+
+    productService.deleteProduct(testProduct.getId());
+
+    assertEquals(0, testProduct.getProductsContent().size());
+  }
+
+  @Test
+  void testDeleteProductSuccessfully() {
+
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+
+    productService.deleteProduct(testProduct.getId());
+
+    assertEquals(0, productRepository.count());
+  }
+
+  @Test
+  void testDeleteProductShouldThrowExceptionWhenProductIsPartOfProduct() {
+    testProduct.setContentOf(new Product());
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+    UUID productId = testProduct.getId();
+    assertThrows(ProductIsContentException.class, () -> productService.deleteProduct(productId));
+  }
+
+  @Test
+  void testDeleteProductShouldThrowExceptionWhenProductIsSold() {
+    testProduct.setSold(true);
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+    UUID productId = testProduct.getId();
+    assertThrows(ProductIsSoldException.class, () -> productService.deleteProduct(productId));
+  }
+
+  @Test
+  void testDeleteProductGetProductShouldThrowExceptionWhenProductNotExist() {
+    UUID fakeId = UUID.randomUUID();
+    assertThrows(ProductNotFoundException.class, () -> productService.deleteProduct(fakeId));
+  }
+
+  @Test
+  void createProductSuccessfully() {
```

Comment:
I am wondering whether we should delete this test as we are testing it in the integration tests.